setup-gcloud/src/setup-gcloud.ts

```diff
@@ -49,15 +49,27 @@ async function run() {
     }
 
     // write the service account key to a temporary file
-    const tmpKeyFilePath = await new Promise<string>((resolve, reject) => {
+    let tmpKeyFilePath = await new Promise<string>((resolve, reject) => {
       tmp.file((err, path, fd, cleanupCallback) => {
         if (err) {
           reject(err);
         }
         resolve(path);
       });
     });
-    await fs.writeFile(tmpKeyFilePath, Base64.decode(serviceAccountKey));
+
+    const serviceAccountFileName = core.getInput('service_account_file_name');
+
+    if (serviceAccountFileName) {
+      tmpKeyFilePath = `/tmp/${serviceAccountFileName}`;
+    }
+
+    if (!fs.existsSync(tmpKeyFilePath)) {
```

Comment:
What about the case when the file does exist? Presumably we'd still want to write the key to the file in-case the key has been changed in the secret store.

Reply:
correct, I'll remove this check